src/miral/basic_window_manager.cpp

```diff
@@ -2993,4 +2981,21 @@ void miral::BasicWindowManager::move_cursor_to(mir::geometry::PointF point)
             sink->handle_input(std::move(event));
         });
     });
-}
\ No newline at end of file
+}
+
+auto miral::BasicWindowManager::window_workspaces_intersect(std::vector<std::shared_ptr<Workspace>> const& w1_workspaces, Window const& w2) const -> bool
+{
+    auto w2_workspaces = workspaces_containing(w2);
```

Comment:
Should `workspaces_containing` just return a sorted list by default?

Reply:
I don't think so. We only needed sorting here because `std::set_intersection` requires that according to the example implementation on [cppreference](https://en.cppreference.com/w/cpp/algorithm/set_intersection)